src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -145,10 +174,17 @@ private void InferParameterBindingSources(ActionModel actionModel)
         // Internal for unit testing.
         internal BindingSource InferBindingSourceForParameter(ParameterModel parameter)
         {
+            var parameterType = parameter.ParameterInfo.ParameterType;
             if (ParameterExistsInAllRoutes(parameter.Action, parameter.ParameterName))
             {
                 return BindingSource.Path;
             }
+            else if (parameterType == typeof(IFormFile) ||
+                parameterType == typeof(IFormFileCollection) ||
+                typeof(IEnumerable<IFormFile>).IsAssignableFrom(parameterType))
+            {
+                return BindingSource.FormFile;
+            }
```

Comment:
Can we do this (infer the binding source) for **all** actions?

Reply:
Sorry, missed this. I'll move this over to the default provider.